src/System.Management.Automation/engine/serialization.cs

```diff
@@ -4039,38 +4053,78 @@ private object ReadDictionary(ContainerType ct)
 
                     object value = ReadOneObject();
 
-                    // On the first collision, copy the hash table to one that uses the default comparer.
-                    if (table.ContainsKey(key) && (keyClashFoundIteration == 0))
+                    if (!isOrdered)
```

Comment:
Why don't swap and start with `if (isOrdered)`?

Reply:
Swapped.